lib/internal/bootstrap/loaders.js

```diff
@@ -143,7 +143,8 @@ internalBinding('module_wrap').callbackMap = new WeakMap();
 const loaderExports = {
   internalBinding,
   NativeModule,
-  require: nativeModuleRequire
+  require: nativeModuleRequire,
+  cjsContext: Object.create(null)
```

Comment:
Is there a reason for `Object.create(null)`? It doesn’t seem to be exposed to userland, and it should make things slower because it puts the object in dictionary mode.

Reply:
Here's an example of why `Object.create(null)` is important:

```js
console.log(hasOwnProperty); // Function
global.hasOwnProperty = 'asdf';
console.log(hasOwnProperty); // 'asdf'
```

with this change, the second `console.log` would output `Function`, since `hasOwnProperty` would be read of the Object prototype from the context.

Reply:
Would `{ __proto__: null, process: undefined, Buffer: undefined }` instead skip the dictionary mode switch?

Reply:
Does this imply that `global`, globalThis, and `Function('return this')()` won’t all be the same object, because there will be a layer wrapped around global? or is there v8 stuff happening here I’m not clear on?

Reply:
No those will all work as expected - the global and all global lookups stay the same. The context lookup only catches `process` and `Buffer` as a minor performance optimization.

Reply:
for future reference: the cjsContext here is implemented by v8 as wrapping the function expression in a with statement.